docs/sources/reference/components/loki/loki.source.file.md

```diff
@@ -105,6 +109,27 @@ If no file changes are detected, the poll frequency doubles until a file change
 
 If file changes are detected, the poll frequency is reset to `min_poll_frequency`.
 
+### `file_match`
+
+The `file_match` block enables built-in file discovery directly within `loki.source.file`.
+When enabled, the component discovers files on the local filesystem using glob patterns and the [doublestar][] library.
```

Comment:
What happens when someone provides file_match and also sends targets in using the previous pattern? 

I think that should ideally be an error, unless there's a simple way to make both work.